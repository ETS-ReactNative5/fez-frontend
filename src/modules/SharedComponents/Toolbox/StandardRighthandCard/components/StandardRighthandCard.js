import React from 'react';
import PropTypes from 'prop-types';
import {HelpIcon} from '../../HelpDrawer';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import Divider from '@material-ui/core/Divider';
import {withStyles} from '@material-ui/core/styles';

const styles = {
    title: {
        minWidth: 1,
    },
    divider: {
        marginTop: 6,
        marginBottom: 12
    }
};

class RighthandCard extends React.Component {
    static propTypes ={
        classes: PropTypes.object,
        children: PropTypes.any,
        title: PropTypes.string,
        help: PropTypes.shape({
            title: PropTypes.string,
            text: PropTypes.any,
            buttonLabel: PropTypes.string
        })
    };

    render() {
        const {classes, title, children, help} = this.props;
        return (
            <Grid container>
                <Grid item xs className={classes.title}>
                    {title &&
                    <Typography variant={'h6'} color={'primary'}>{title}</Typography>}
                </Grid>
                {help && help.text &&
                <Grid item>
                    <HelpIcon {...help}/>
                </Grid>
                }
                <Grid item xs={12}>
                    <Divider className={classes.divider} />
                </Grid>
                <Grid item xs={12}>
                    {children}
                </Grid>
            </Grid>
        );
    }
}

const StyledCard = withStyles(styles, {withTheme: true})(RighthandCard);
const StandardRighthandCard = (props) => <StyledCard {...props}/>;
export default StandardRighthandCard;
