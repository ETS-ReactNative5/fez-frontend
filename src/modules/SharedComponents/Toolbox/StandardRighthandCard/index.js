export { default as StandardRighthandCard } from './components/StandardRighthandCard';
