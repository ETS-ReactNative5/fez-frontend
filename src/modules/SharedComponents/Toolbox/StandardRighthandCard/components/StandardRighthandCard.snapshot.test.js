import StandardRighthandCard from './StandardRighthandCard';

function setup(testProps, isShallow = true) {
    // build full props list required by the component
    const props = {
        ...testProps,
        title: testProps.title,
        help: testProps.help,
        children: testProps.children
    };
    return getElement(StandardRighthandCard, props, isShallow);
}

describe('Snapshot tests for StandardRighthandCard component', () => {
    it('renders StandardRighthandCard with title and no help icon', () => {
        const wrapper = setup({title: 'card title'});
        const tree = toJson(wrapper);
        expect(tree).toMatchSnapshot();
    });

    it('renders StandardRighthandCard with title and help button', () => {
        const wrapper = setup({title: 'Title', help: {title:'Help text', text:'Some help text', buttonLabel:'OK'}});
        const tree = toJson(wrapper);
        expect(tree).toMatchSnapshot();
    });

    it('renders StandardRighthandCard with title and help button and some content', () => {
        const wrapper = setup({title: 'Title', help: {title:'Help text', text:'Some help text', buttonLabel:'OK'}, children: 'Some content'});
        const tree = toJson(wrapper);
        expect(tree).toMatchSnapshot();
    });
});
