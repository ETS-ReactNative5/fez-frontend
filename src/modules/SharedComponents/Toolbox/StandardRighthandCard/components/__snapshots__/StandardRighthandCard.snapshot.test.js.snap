// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests for RighthandCard component renders with title and help button 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={true}
  >
    <WithStyles(ForwardRef(Typography))
      color="primary"
      variant="h6"
    >
      Title
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <Connect(HelpIcon)
      buttonLabel="OK"
      text="Some help text"
      title="Help text"
    />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={12}
  >
    <WithStyles(ForwardRef(Divider)) />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={12}
  />
</WithStyles(ForwardRef(Grid))>
`;

exports[`Snapshot tests for RighthandCard component renders with title and help button and some content 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={true}
  >
    <WithStyles(ForwardRef(Typography))
      color="primary"
      variant="h6"
    >
      Title
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <Connect(HelpIcon)
      buttonLabel="OK"
      text="Some help text"
      title="Help text"
    />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={12}
  >
    <WithStyles(ForwardRef(Divider)) />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={12}
  >
    Some content
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;

exports[`Snapshot tests for RighthandCard component renders with title and no help icon 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={true}
  >
    <WithStyles(ForwardRef(Typography))
      color="primary"
      variant="h6"
    >
      card title
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={12}
  >
    <WithStyles(ForwardRef(Divider)) />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={12}
  />
</WithStyles(ForwardRef(Grid))>
`;
